### PR TITLE
Chinese README

### PR DESCRIPTION
**Pull request by Endermanbugzjfc**

### CODE REVIEW

**Comment by SOF3** [comment]
why did you replace the original content instead of creating a new file?

File: README.md [hunk 1]
```diff
@@ -1,17 +1,19 @@
+Eng | [繁](README.zho.md) | [簡](README.chs.md)
```
Comment (SOF3) [inline README.md]: simplified doesn't exist yet right?

Comment (SOF3) [reply README.md]: that'd mean one of the two has to become a source of truth. Does that sound like a good idea?

Comment (SOF3) [reply README.md]: @Endermanbugzjfc it just feels too complex to setup a workflow just for translating the document.

File: zho/README.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    2 => "Timeout!\n",
+};
+```
+
+## 使用後的代碼可以維持「backward compatibility 回溯相容性」嗎？
+是的，  await-generator 不會對已有的程式接口（API）造成任何限制。
+你可以將所有涉及 await-generator 的代碼封閉在應用程式的內部。
+但你確實應該把 generation function 直接當作程序接口。
+
+await-generator 會在 `Await::f2c` method 開始進行異步代碼流控制，它接受一般的回調語法，這樣就無需再 <!-- TODO: help wanted-->
```
Comment (SOF3) [inline zho/README.md:88]: `Await::f2c` doesn't accept callback syntax. I am just saying that you can use this wrapper to create callback style API.

File: zho/README.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+```
+
+傳統「回調式」的異步 function 也可以被 `Await::promise` method 轉化成「等待式」。它跟 JavaScript 的 `new Promise` 很像：
+
+```php
+yield from Await::promise(fn($resolve, $reject) => oldFunction($args, $resolve, $reject));
+```
+
+## await-generator 的*劣勢*
+await-generator 也有很多經常的坑人的地方：
```
Comment (SOF3) [inline zho/README.md:119]:
```suggestion
await-generator 也有很多經常坑人的地方：
```

File: zho/README.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+- 忘了 `yield from Generator<void>` 的結果是代碼會毫無作用；
```
Comment (SOF3) [inline zho/README.md:121]:
```suggestion
- 忘了 `yield from` 的代碼會毫無作用；
```

File: zho/README.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+- 如果你的 function 沒有任何 `yield` 或者 `yield from` ， PHP 就不會把它當成 generator function 。（將所有 generator function 的 return 類型設成 `: Generator` 可減輕影響）；
+- `finally` blocks may never get executed if an async function never resolves
+  (e.g. `Await::promise(fn($resolve) => null)`)；
```
Comment (SOF3) [inline zho/README.md:124]: delete these two lines?

File: zho/README.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+- 如果異步代碼沒有全面結束， `finally` 也不會被執行 （例： `Await::promise(fn($resolve) => null)`）；
+
+儘管地方會導致一些問題， await-generator 的設計模式依然比「回調地獄」更難出 bug 。
```
Comment (SOF3) [inline zho/README.md:127]:
```suggestion
儘管一些地方會導致問題， await-generator 的設計模式出 bug的機會依然比「回調地獄」少 。
```

**Review by SOF3 — CHANGES_REQUESTED** [review]

**Comment by Endermanbugzjfc** [comment]
Btw line breaks will cause space in the middle of sentences. Shall we make the sentence format uniform? Maybe one space after each full stop?

File: zho/README.md [hunk 7]
```diff
@@ -133,20 +133,16 @@ await-generator 也有很多經常坑人的地方：
...
-wouldn't call functions that modify the private states of your library.
-But this assumption does not hold with fibers
-because the fiber is preempted and other fibers can still modify the private states.
-This means you have to check for possible changes in private properties
-every time you call any function that *might* be suspending.
-
-On the other hand, using explicit await,
-it is obvious where exactly the suspension points are,
-and you only need to check for state mutations at the known suspension points.
+### 連貫的狀態
```
Comment (SOF3) [inline zho/README.md:136]:
```suggestion
### 並發狀態
```

File: zho/README.md [hunk 8]
```diff
@@ -133,20 +133,16 @@ await-generator 也有很多經常坑人的地方：
...
+當一個函數被暫停時會發生許多其他的事情。
+的確，這樣在呼叫它時給予了其他執行者呼叫任何可修改你狀態的函數。
```
Comment (SOF3) [inline zho/README.md:138]:
```suggestion
調用函數時固然給予了實現者調用可修改狀態的函數的可能性，
```

File: zho/README.md [hunk 9]
```diff
@@ -133,20 +133,16 @@ await-generator 也有很多經常坑人的地方：
...
+但是一個正常的、合理的構造，例如 HTTP 請求所呼叫的函數不應會修改你程式庫的私有狀態。
```
Comment (SOF3) [inline zho/README.md:139]:
```suggestion
但是一個正常的、合理的實現，例如 HTTP 請求所調用的函數不應修改你程式庫的私有狀態。
```

File: zho/README.md [hunk 10]
```diff
@@ -133,20 +133,16 @@ await-generator 也有很多經常坑人的地方：
...
+但是這個假設對於 fibers 來說並不成立
+因為當一個光纖被暫停後，剩下的仍然可以修改內部狀態。
```
Comment (SOF3) [inline zho/README.md:141]:
```suggestion
因為當一個纖程被暫停後，其他纖程仍然可以修改你的內部狀態。
```

File: zho/README.md [hunk 11]
```diff
@@ -133,20 +133,16 @@ await-generator 也有很多經常坑人的地方：
...
-every time you call any function that *might* be suspending.
-
-On the other hand, using explicit await,
-it is obvious where exactly the suspension points are,
-and you only need to check for state mutations at the known suspension points.
+### 連貫的狀態
+當一個函數被暫停時會發生許多其他的事情。
+的確，這樣在呼叫它時給予了其他執行者呼叫任何可修改你狀態的函數。
+但是一個正常的、合理的構造，例如 HTTP 請求所呼叫的函數不應會修改你程式庫的私有狀態。
+但是這個假設對於 fibers 來說並不成立
```
Comment (SOF3) [inline zho/README.md:140]:
```suggestion
但是這個假設對於 fibers 來說並不成立，
```

File: zho/README.md [hunk 12]
```diff
@@ -133,20 +133,16 @@ await-generator 也有很多經常坑人的地方：
...
+因為當一個光纖被暫停後，剩下的仍然可以修改內部狀態。
+每次你呼叫任何*可能*會被暫停的函數時，你都必須檢查內部屬性的可能變化。
```
Comment (SOF3) [inline zho/README.md:142]:
```suggestion
每次你調用任何*可能*會被暫停的函數時，你都必須檢查內部狀態的可能變化。
```

**Review by SOF3 — CHANGES_REQUESTED** [review]

**Comment by SOF3** [comment]
Please standardize a few terms

Fibers => 纖程 (not optical fibers!!! -.-)
implementation => 實現
function call => 函數調用

File: zho/README.md (hunk 8 above)
Comment (Endermanbugzjfc) [reply zho/README.md:138]: > Suggested change
> 的確，這樣在呼叫它時給予了其他執行者呼叫任何可修改你狀態的函數。
> 調用函數時固然給予了實現者調用可修改狀態的函數的可能性，

```diff
- 狀態的函數的可能性
+ 狀態函數的可能性
```
Two `的`s.

File: zho/README.md (hunk 9 above)
Comment (Endermanbugzjfc) [reply zho/README.md:139]: Oh I forgot to change 私有狀態 to 內部狀態.

File: zho/README.md (hunk 8 above)
Comment (Endermanbugzjfc) [reply zho/README.md:138]:
```suggestion
調用函數時固然給予了實現者調用可修改狀態函數的可能性，
```

File: zho/README.md (hunk 9 above)
Comment (Endermanbugzjfc) [reply zho/README.md:139]:
```suggestion
但是一個正常的、合理的實現，例如 HTTP 請求所調用的函數不應修改你程式庫的內部狀態。
```

File: zho/README.md (hunk 7 above)
Comment (Endermanbugzjfc) [reply zho/README.md:136]: Why was this suggestion block not showing up in the Files tab? And 並發狀態 sounds so abstract...

**Comment by Endermanbugzjfc** [comment]
> Fibers => 纖程 (not optical fibers!!! -.-)
Blame DeepL heh.

**Comment by Endermanbugzjfc** [comment]
How to mute this bot?

**Comment by Endermanbugzjfc** [comment]
@SOF3 I think a `book-translation` branch should be created. Translations live in that branch while the English version will still be kept on master.

**Comment by Endermanbugzjfc** [comment]
I will create another PR for the book translations.

**Review by BluSimmon — APPROVED** [review]

**Review by BluSimmon — APPROVED** [review]

**Comment by Endermanbugzjfc** [comment]
I will not translate the code in README.md nor to add explaination comments until someone complains about this.

File: zho/README.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+    2 => onTimeout(),
+])
+echo match($which) {
+    0 => "Done!\n",
+    1 => "User closed!\n",
+    2 => "Timeout!\n",
+};
+```
+
+## 使用後的代碼可以維持「backward compatibility 回溯相容性」嗎？
```
Comment (SOF3) [inline zho/README.md:83]:
```suggestion
## 使用後的代碼可以維持回溯相容性嗎？
```

no need to keep the english term

File: zho/README.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+是的， await-generator 不會對已有的程式接口（API）造成任何限制。
```
Comment (SOF3) [inline zho/README.md:84]:
```suggestion
是的， await-generator 不會對已有的接口造成任何限制。
```

File: zho/README.md [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+你可以將所有涉及 await-generator 的代碼封閉在程式的內部。
+但你確實應該把生成器函數直接當作程式接口。
+
+await-generator 會在 `Await::f2c` method 開始進行異步代碼流控制，你可以將它視為「等待式」至「回調式」的轉接頭。
```
Comment (SOF3) [inline zho/README.md:88]:
```suggestion
await-generator 會在 `Await::f2c` 開始進行異步代碼流控制，你可以將它視為「等待式」至「回調式」的轉接頭。
```

File: zho/README.md [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+- 如果異步代碼沒有全面結束， `finally` 裏面的代碼也不會被執行（例： `Await::promise(fn($resolve) => null)`）；
+
+儘管一些地方會導致問題， await-generator 的設計模式出 bug 的機會依然比「回調地獄」少 。
+
+## 不是有纖程嗎？
+雖然這樣說很主觀，但本人因為以下纖程缺少的特色而相對地不喜歡它：
+
+### 靠類型註釋就能區分異步、非異步函數
+> 先生，你已在暫停的纖程待了三十秒。<br />
+> 因為有人為了實現一個界面而調用 Fiber::suspend() 。
```
Comment (SOF3) [inline zho/README.md:131]:
```suggestion
> 因為有人實現一個界面時調用了 `Fiber::suspend() ` 。
```

File: zho/README.md [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+
+![../../fiber.jpg](https://github.com/SOF3/await-generator/raw/master/fiber.jpeg)
+
+> 好傢伙，我都等不及要回應我的 HTTP 請求了。<br />
+> 那個框架肯定還沒把它給清除。
```
Comment (SOF3) [inline zho/README.md:136]:
```suggestion
> 框架肯定還沒把它給超時清除。
```

**Review by SOF3 — CHANGES_REQUESTED** [review]
Please also update the corresponding changes for chs

**Comment by Endermanbugzjfc** [comment]
Sorry, I have not yet figure out how to batch suggestions on mobile app...

**Comment by Endermanbugzjfc** [comment]
1 sec,

**Comment by Endermanbugzjfc** [comment]
> Changes requested
What changes? I committed all.

**Review by SOF3 — APPROVED** [review]

**Pull request merged**